src/core/public/plugins/plugin.ts

```diff
@@ -47,8 +47,8 @@ export interface Plugin<
 export type PluginInitializer<
   TSetup,
   TStart,
-  TPluginsSetup extends Record<string, unknown> = {},
```

Comment:
@elastic/kibana-platform This caused issues as it states that the type for `TPluginsSetup` must include an index signature, like `{ [key: string]: unknown }`, which doesn't seem like what was intended. Do you think basing this type on `{}` will cause issues?

Reply:
`{}` means an empty interface and any data type is valid type
```typescript
type A<T> = T extends {} ? 'foo' : 'bar';

type B = A<string>; // foo
type C = A<number>; // foo
type D = A<{}>;     // foo
type E = A<void>;   // bar
```

if we use `Record<string, any>` or `object` we will have a problem that any non-primitive type is valid one
```typescript
type A<T> = T extends Record<string, any> ? 'foo' : 'bar';

type B = A<string>; // bar
type C = A<number>; // bar
type D = A<{}>;     // foo
type E = A<void>;   // bar
type F = A<() => null>; // foo
type G = A<string[]>;   // foo
```

could we switch to the solution that you proposed in DM
`TPluginsSetup extend { [key: string]: unknown }` ?

Reply:
Unfortunately `TPluginsSetup extend { [key: string]: unknown }` is functionally equivalent to `Record<string, unknown>`, it still requires that every class trying to implement `PluginInitializer` must define their `TPluginsSetup` type to include the `[key: string]: unknown` index signature.

Reply:
Should we use `Record<string, any>` or `object` instead of `{}` in this case? At least we specify `TPluginsSetup` type more precisely

Reply:
`Record<string, any>` works, I'll go for that. Thanks